numpy/ma/core.py

```diff
@@ -3351,6 +3354,8 @@ def __setattr__(self, attr, value):
                 self._mask.shape = self.shape
             except (AttributeError, TypeError):
                 pass
+        elif attr == 'shape' and getmask(self) is not nomask:
```

Comment:
Might be clearer to add `dtype` and `shape` `@properties` that fall back on `ndarray.shape.__get__` etc